### ISSUE TITLE
Warmup epochs with frozen pre-trained encoder weights to initialize decoder

### ISSUE THREAD

**Issue by daniel-j-h** [mapbox/robosat#72]
At the moment we are using a pre-trained ResNet as an encoder in our encoder-decoder architecture:

https://github.com/mapbox/robosat/blob/8b7566e3b5de3583eb64fbb641ff18b9ff01f901/robosat/unet.py#L94-L100

https://github.com/mapbox/robosat/blob/8b7566e3b5de3583eb64fbb641ff18b9ff01f901/robosat/unet.py#L123-L134

We are currently training the model as is with all layers unfrozen.

We should investigate if freezing the ResNet encoder and running a few warmup epochs to initialize the decoder layers (then unfreezing parts or all of the ResNet) helps.

Here is how we can freeze the encoder - unfreezing works similarly:

```python3
def freeze(self):
    for layer in (self.enc0, self.enc1, self.enc2, self.enc3, self.enc4):
        for param in layer.parameters():
            param.requires_grad = False
```
